cfme/common/__init__.py

```diff
@@ -420,6 +420,9 @@ def add_tag(self, tag=None, cancel=False, reset=False, details=True, dashboard=F
                             default is details page
             dashboard (bool): Set to True if tag should be added via the Dashboard view
         """
+        if tag and tag in self.get_tags():
+            logger.warning('Trying to add tag [%s] already assigned to entity [%s]', tag, self)
```

Comment:
I have a concern here but I haven't tested. I doubt it will return True because I remember examples in our framework when `tag in get_tags()` doesn't work.

For example,
```
    tags = job_template.get_tags()
    msg = "Failed to setup a configuration system's tag"
    tag_info = '{}: {}'.format(tag.category.display_name, tag.display_name)
    assert tag_info in ['{}: {}'.format(t.category.display_name, t.display_name) for t in tags], msg
```

Reply:
nevermind, I see you updated that method as well :)

Reply:
yep!

This should now work with the update to `get_tags` and the `Tag.__eq__` method, and I did some cursory testing of `in`, `not in`, `==` comparisons of tag objects.

Reply:
I'll update `test_config_management.py` - I searched for references to `tag.name` that were expecting that to be set to the category display_name, but didn't locate that one.

Reply:
https://github.com/ManageIQ/integration_tests/pull/9012/commits/4bc2d023e0d7936ac327677f7df843dbe218678e